evm/evm.go

```diff
@@ -144,12 +140,14 @@ type Evm struct {
 	chainConfig     params.ChainConfig
 	vmConfig        vm.Config
 	validateTxValue bool
+	gasLimit        uint64
 }
 
 func NewEvm(sdb vm.StateDB, lstate loomchain.State, abm *evmAccountBalanceManager, debug bool) *Evm {
 	p := new(Evm)
 	p.sdb = sdb
-
+	onChainCfg := lstate.Config()
+	p.gasLimit = onChainCfg.Evm.GasLimit
```

Comment:
Probably need to do `onChainCfg.GetEvm().GasLimit` because `onChainCfg.Evm` is likely to be `nil`.
And need a feature flag.